### DISCUSSION TITLE
코드에 주석을 다는 것에 대해서

### DISCUSSION THREAD

**Discussion by rlajm1203** [JNU-econovation/EEOS-BE#171] [General]
EEOS 코드에서 급하게 수정해야 할 부분이 있어서 코드를 읽던 중 궁금함이 생겨서 디스커션을 남깁니다.

1. 여러분들 께서는 코드를 작성하실 때 주석을 다는 것에 대해서 어떻게 생각하시나요?

### 벨민의 생각
저는 주석을 웬만하면 작성하는 편이 좋다고 생각합니다.

그 이유는, 가장 1순위로 협업 때문입니다. 제가 EEOS 코드를 처음 읽었을 때 EEOS가 내부적으로 동작하는 방식을 오로지 코드 분석을 통해서만 파악해야 했습니다.

그리고 그렇게 코드만으로 시스템을 파악하는 데에 적지 않은 시간이 걸렸습니다.

그래서 저는 다른 사람과 협업을 하거나 또는 인수인계 과정을 거쳐야 할 경우에 주석을 작성하는 것이 좋다는 생각이 듭니다.

그리고 이 디스커션을 작성하는 지금도 느끼지만, 오늘 EEOS SelectAttendCommandTargetMemberMemberService 의 코드를 보는데
예전에는 분명 이해 했었던 것 같은데, 지금 다시보니 집중해서 읽어야 이해를 할 수 있습니다. 

이렇듯, 이전에 이해했던 코드도 이후에 다시 보면 이해가 안되는 경우도 있습니다.

그리고, 코드를 작성하다가 주석을 작성하는 습관을 들이면, 내가 현재 하고 있는 생각을 제때 기록할 수 있어 좋다고 생각합니다.

이를 github에 푸시하면 팀원이 나의 생각을 그대로 읽을 수도 있고요.

그래서 저는 주석을 웬만하면 작성하는 편이 좋다고 생각합니다. 또한 현재도 다른 프로젝트에서 그렇게 하고 있습니다.

### 우려사항
하지만, 실제 회사같은 경우에는 회사 내부 규칙으로 인해 작성한 코드에 주석을 모두 제거하라고 할 수 있습니다.

보안 때문이겠지요. 이러한 상황은 어쩔 수 없는 것 같습니다. 회사에서 팀끼리 협업할 때도 주석을 작성하지 말라고 할 수 있을 것 같습니다.

산업 스파이 때문에요. (그냥 그럴 것 같다는 저의 근거 없는 생각..)

여러분들께서는 어떻게 생각하시나요?

**Comment by kssumin** [comment]
## 주석보단 코드 퀄리티가 우선이다.
저는 우선은 주석이 없어도 이해할 수 있는 코드를 짜는 게 젤 중요한 것 같다고 생각합니다.
사실 코드가 이해가 안 된다는 건 그만큼 코드를 더럽게.. 짰기 때문..(SelectAttendCommandTargetMemberMemberService 짠 사람 누구? 나 ㅋ)
그래서 더러운 코드를 짜도 주석을 통해 팀원간 이해가 가능하게 한다.로 간다면 저는 반대입니다.

## 코드 퀄리티도 챙기면서 주석으로 보충을 한다.
만약 위 같은 의견이라면 주석도 Javadoc 규칙에 따라서 팀원간의 공통된 규칙을 바탕으로 작성하면 찬성합니다!

(Javadoc 관련해서는 이펙티브 자바나 관련 블로그 보셔도 될 것 같습니다!)
